lib/client/ai_client.dart

```diff
@@ -0,0 +1,77 @@
+import 'dart:async';
+
+import 'package:app/core/config.dart';
+import 'package:app/core/oauth/oauth.dart';
+import 'package:app/injection/register_module.dart';
+import 'package:ferry/ferry.dart';
+import 'package:ferry_hive_store/ferry_hive_store.dart' as ferry_hive_store;
+import 'package:graphql_flutter/graphql_flutter.dart';
+import 'package:hive_flutter/adapters.dart';
+import 'package:injectable/injectable.dart';
+
+@LazySingleton()
+class AIClient {
+  late Client _client;
+  late WebSocketLink _webSocketLink;
+  late AuthLink _authLink;
+  late final HttpLink _httpLink = HttpLink(AppConfig.aiUrl);
+  late final StreamSubscription<OAuthTokenState> tokenStateSubscription;
+
+  Client get client => _client;
+
+  init() async {
```

Comment:
we can put these logic inside constructor function call, so that we won't need to call init() in main.dart file